scipy/signal/tests/test_filter_design.py

```diff
@@ -3695,17 +3732,17 @@ def test_frequency_response(self):
         # Check if the frequency response fulfill the specifications:
         # hp[0] and hp[4]  correspond to frequencies distant from
         # w0 = 0.3 and should be close to 1
-        assert_allclose(abs(hp[0]), 1, rtol=1e-2)
-        assert_allclose(abs(hp[4]), 1, rtol=1e-2)
+        xp_assert_close(abs(hp[0]), 1., rtol=1e-2, check_0d=False)
+        xp_assert_close(abs(hp[4]), 1., rtol=1e-2, check_0d=False)
```

Comment:
When we add other libraries, we'll be expecting an array here from `xp.abs`. So we can either use `np.asarray(1.)[()]` (which I know you hate 😅) or `from scipy._lib._array_api_no_0d import xp_assert_close as xp_assert_close_no_0d` to get the 0d check here.

Reply:
No, I will not add `[()]` here or elsewhere. How about we keep working things working and see what actually needs to change when non-numpy arrays actually appear?

Reply:
Okay, we can leave it like this for now and change it at a later date instead as you wish.